src/Peering/Peering/Direct/SetAzureDirectPeeringConnectionCommand.cs

```diff
@@ -176,14 +176,14 @@ public override void Execute()
         /// </returns>
         private PSDirectConnection UpdatePeeringOffer()
         {
-            if (this.InputObject is PSDirectConnection inputObject)
+            if (this.InputObject.GetType().Equals(typeof(PSDirectConnection)))
```

Comment:
Why use direct type comparsion instead of `is`? Except for extremely rare cases, `is` is what you want.

Reply:
This was causing an unexpected response, because the model goes through a conversion from SDK model -> PS model -> PSViewModel <- and back.

Reply:
As per my Teams message, `InputObject` is declared as `PSDirectConnection`. So, the type check (in either case) doesn't make sense.